### PR TITLE
[CatalogPromotion] batch configuration size made a part of config

### PR DESCRIPTION
**Pull request by TheMilek**
| Q               | A
| --------------- | -----
| Branch?         | 1.11
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no 
| License         | MIT

In case of better management of batch sizes we decided to extract it to CatalogPromotion config, then we can resize it by modifying `_sylius.yaml` file

### CODE REVIEW

File: config/packages/_sylius.yaml [hunk 1]
```diff
@@ -16,3 +16,7 @@ sylius_api:
 sylius_shop:
     product_grid:
         include_all_descendants: true
+
+sylius_core:
```
Comment (lchrusciel) [inline config/packages/_sylius.yaml:20]: Should be removed

File: src/Sylius/Bundle/CoreBundle/Tests/DependencyInjection/SyliusCoreExtensionTest.php [hunk 2]
```diff
@@ -68,6 +68,36 @@ public function it_does_not_autoconfigure_prepending_doctrine_migrations_if_it_i
...
 
+    /**
+     * @test
+     */
+    public function it_loads_batch_size_parameter_value_to_dev_container_properly(): void
+    {
+        $this->configureContainer('dev');
+
+        $this->assertIfBatchParameterIsLoaded();
+    }
```
Comment (lchrusciel) [inline src/Sylius/Bundle/CoreBundle/Tests/DependencyInjection/SyliusCoreExtensionTest.php:79]: Requires refactoring ti check default and configured values

File: src/Sylius/Bundle/CoreBundle/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -51,6 +51,12 @@ public function getConfigTreeBuilder(): TreeBuilder
                 ->scalarNode('driver')->defaultValue(SyliusResourceBundle::DRIVER_DOCTRINE_ORM)->end()
                 ->booleanNode('prepend_doctrine_migrations')->defaultTrue()->end()
                 ->booleanNode('shipping_address_based_taxation')->defaultFalse()->end()
+                ->arrayNode('applying_catalog_promotion_on_variants')
```
Comment (Zales0123) [inline src/Sylius/Bundle/CoreBundle/DependencyInjection/Configuration.php:54]: I'm wondering... why not just `catalog_promotions`? 😄 this is obvious we do not apply catalog promotions on anything else than variants and even if one would like to do it - the configuration itself can be used in multiple contexts, not only for variants 💃

File: src/Sylius/Bundle/CoreBundle/DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -51,6 +51,12 @@ public function getConfigTreeBuilder(): TreeBuilder
...
+                    ->addDefaultsIfNotSet()
+                    ->children()
+                        ->integerNode('size_of_batches')->defaultValue(100)->end()
```
Comment (Zales0123) [inline src/Sylius/Bundle/CoreBundle/DependencyInjection/Configuration.php:57]: `batch_size`?

File: src/Sylius/Bundle/CoreBundle/DependencyInjection/Configuration.php (hunk 3 above)
Comment (Zales0123) [reply src/Sylius/Bundle/CoreBundle/DependencyInjection/Configuration.php:54]: Additionally - there would be no verb in the configuration, which means better consistency

File: src/Sylius/Bundle/CoreBundle/spec/CommandDispatcher/BatchedApplyCatalogPromotionsOnVariantsCommandDispatcherSpec.php [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+use Symfony\Component\Messenger\MessageBusInterface;
+
+final class BatchedApplyCatalogPromotionsOnVariantsCommandDispatcherSpec extends ObjectBehavior
+{
+    public function let(MessageBusInterface $messageBus,): void
+    {
+        $this->beConstructedWith($messageBus, 2);
+    }
+
+    function it_implements_its_interface(): void
```
Comment (Zales0123) [inline src/Sylius/Bundle/CoreBundle/spec/CommandDispatcher/BatchedApplyCatalogPromotionsOnVariantsCommandDispatcherSpec.php:29]: `it_implements_apply_catalog_promotions_on_variants_command_dispatcher_interface` 🔥 I know it's long, but spec steps should be precise

File: src/Sylius/Bundle/CoreBundle/spec/CommandDispatcher/BatchedApplyCatalogPromotionsOnVariantsCommandDispatcherSpec.php [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    {
+        $this->shouldImplement(ApplyCatalogPromotionsOnVariantsCommandDispatcherInterface::class);
+    }
+
+    public function it_dispatches_in_two_batches(MessageBusInterface $messageBus): void
+    {
+        $variantCodes = ['example_variant1', 'example_variant2', 'example_variant3'];
```
Comment (Zales0123) [inline src/Sylius/Bundle/CoreBundle/spec/CommandDispatcher/BatchedApplyCatalogPromotionsOnVariantsCommandDispatcherSpec.php:36]: I would skip the variable and just call `$this->updateVariants(['example_variant1', 'example_variant2', 'example_variant3']);` in line 44. Similar comment to the spec below

File: src/Sylius/Bundle/CoreBundle/spec/CommandDispatcher/BatchedApplyCatalogPromotionsOnVariantsCommandDispatcherSpec.php [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+        $twoVariantCodes = new ApplyCatalogPromotionsOnVariants(['example_variant1', 'example_variant2']);
```
Comment (Rafikooo) [inline src/Sylius/Bundle/CoreBundle/spec/CommandDispatcher/BatchedApplyCatalogPromotionsOnVariantsCommandDispatcherSpec.php:38]:
```suggestion
        $firstCommand = new ApplyCatalogPromotionsOnVariants(['example_variant1', 'example_variant2']);
```
Imho `$twoVarianCodes` variable name is misleading as it contains an object

File: src/Sylius/Bundle/CoreBundle/spec/CommandDispatcher/BatchedApplyCatalogPromotionsOnVariantsCommandDispatcherSpec.php (hunk 5 above)
Comment (lchrusciel) [reply src/Sylius/Bundle/CoreBundle/spec/CommandDispatcher/BatchedApplyCatalogPromotionsOnVariantsCommandDispatcherSpec.php:29]: My favorit:
```suggestion
    function it_is_apply_catalog_promotions_on_variants_command_dispatcher(): void
```

**Review by Zales0123 — APPROVED** [review]

**Review by lchrusciel — APPROVED** [review]

**Comment by lchrusciel** [comment]
Thanks, Kamil! :tada:

**Pull request merged**